lib/vehicle_factory.rb

```diff
@@ -0,0 +1,38 @@
+require './vehicle'
+
+class VehicleFactory
+  attr_accessor :created_vehicles
+  
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles_by_state(state_database)
+    wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    ny_state_registrations = DmvDataService.new.ny_dmv_office_locations
+
+    if state_database == wa_ev_registrations
```

Comment:
I don't love the direct comparison to `DmvDataService.new.wa_ev_registrations` (and same for New York) because that is going to execute the creation of a `DmvDataService` object AND make the API call to make sure the two data sets match. This isn't ideal for two reasons.

1. It's expensive to make API calls. It's really slow. You probably noticed your test suite slowing down as you progressed through the project. It can also literally be expensive. A lot of APIs cost money. You pay every time you access the API's data. That's not the case here, but often is.
2. While not super likely, it's possible that data changes between the time you call your `create_vehicles_by_state` method and lines 14 or 25 execute which means it could actually be Washington EV Registration data but since the data sets don't 100% match, it'll return false.

Instead I would approach this by finding something ABOUT the data you can match. For example is there a key/value pair in the hash that is unique to the Washington data set?